src/compile/data/bin.ts

```diff
@@ -89,6 +90,38 @@ export class BinNode extends DataFlowNode {
     return new BinNode(bins);
   }
 
+  public static makeTransform(model: ModelWithField, t: BinTransform) {
+    const bins: Dict<BinComponent> = {};
+    model.forEachFieldDef((fieldDef, channel) => {
```

Comment:
Why is this calling `forEachFieldDef`, which basically enumerates `encoding`?  

This bin in the transform basically should be orthogonal to the encoding.